homework-submission/react-week3-v2/src/HomeworkWeek3/HomeWorkWeek3.js

```diff
@@ -0,0 +1,79 @@
+import React, { useState, useEffect, useCallback } from "react";
+import UserCard from "../components/UserCard";
+import UserInfo from "../components/UserInfo/UserInfo";
+import Spinner from "../components/Spinner";
+import Button from "../components/Button";
+import "./HomeworkWeek3.css";
+
+function HomeWorkWeek3() {
+  const API = "https://uinames.com/api/?ext";
+  const [data, setData] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    async function fetchData() {
+      setIsLoading(true);
+      const res = await fetch(API);
+      const usersData = await res.json();
+      setIsLoading(false);
+      setData(usersData);
+    }
+    fetchData();
+
+    return function cleanup() {
+      return fetchData();
+    };
+  }, []);
+
+  const onHandleClick = useCallback(() => {
+    async function fetchNewUser() {
+      setIsLoading(true);
+      const res = await fetch(API);
+      const newUserData = await res.json();
+      setIsLoading(false);
+      setData(newUserData);
+    }
+    fetchNewUser();
+
+    return function cleanup() {
+      return fetchNewUser();
+    };
+  }, []);
+
+  document.body.onkeyup = function(e) {
+    if (e.keyCode === 32) {
+      async function fetchWithSpace() {
+        setIsLoading(true);
+        const res = await fetch(API);
+        const newUserData = await res.json();
+        setIsLoading(false);
+        setData(newUserData);
+      }
+      return fetchWithSpace();
+    }
+  };
```

Comment:
Ah and here is another time almost the same implementation, try to abstract code to a function or to combine the three